### PR TITLE
Add Dashboard Sharing feature tour

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #5328 

## Relevant technical choices

This PR adds a feature tour following the existing feature tours targeting the Dashboard Sharing settings button.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/feature-tours/dashboard-sharing.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Dashboard Sharing Feature Tour.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (kuasha420) [inline assets/js/feature-tours/dashboard-sharing.js:4]: We are getting old and it's already half way in 2022.

**Review by kuasha420 — CHANGES_REQUESTED** [review]
Nice work @nfmohit . Everything looks good and the tour is showing up correctly. Just one issue regarding the copyright year which I've highlighted below.

File: assets/js/feature-tours/dashboard-sharing.js (hunk 1 above)
Comment (kuasha420) [reply assets/js/feature-tours/dashboard-sharing.js]: No worries! We all make this mistake. :dagger:

**Review by kuasha420 — APPROVED** [review]
@nfmohit LGTM! Thanks.

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**